### PR TITLE
ci: Add PR title check

### PR DESCRIPTION
**Pull request by rohan-bes**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -163,3 +163,46 @@ In order to automatically format code in VS Code according to our style guide:
 2. Enable the **Editor: Format on Save** setting: `"editor.formatOnSave": true`.
 
 Your files will now be formatted automatically when you save them.
+
+### PR title rules
```
Comment (jaskfla) [inline README.md:167]: Does this addition belong in the `CONTRIBUTING.md` instead?

We also have a (public!) Slab doc: https://beyond-essential.slab.com/posts/pr-titles-conventional-commits-avgsj3xb

File: README.md [hunk 2]
```diff
@@ -163,3 +163,46 @@ In order to automatically format code in VS Code according to our style guide:
...
+
+We use a squash-and-merge strategy when merging PRs into dev. This means the commit messages in dev will match the PR titles. We like to keep standardised commit messages as an easy way to track release changes and for record keeping purposes.
+
+The title of pull requests must be in [Conventional Commit] format. This is used to generate
```
Comment (jaskfla) [inline README.md:171]:
```suggestion
The title of pull requests must be in [Conventional Commit](https://beyond-essential.slab.com/posts/pr-titles-conventional-commits-avgsj3xb) format. This is used to generate
```

File: README.md [hunk 3]
```diff
@@ -163,3 +163,46 @@ In order to automatically format code in VS Code according to our style guide:
...
+changelogs, and to provide a consistent format for commits landing in the main branch, as pull
+requests are merged in "squash" mode.
```
Comment (jaskfla) [inline README.md:173]:
```suggestion
requests are merged in “squash” mode.
```

Admittedly a nitpick of nitpicks

File: .github/workflows/check-pr-title.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+          if ! grep -qP '^\w+(\(\w+\))?:\s' <<< "$PR_TITLE"; then
+            echo "::warning::PR title should be in Conventional Commit style, e.g. 'feat: ...'"
+            exit 1
+          fi
+
+      - name: Check for conventional type allow-list
+        if: steps.release.conclusion == 'skipped'
+        shell: bash
+        run: |
+          if ! grep -qP '^(ci|db|deps|doc|env|feat|fix|fmt|merge|refactor|repo|revert|style|test|tweak)(\(\w+\))?:\s' <<< "$PR_TITLE"; then
```
Comment (jaskfla) [inline .github/workflows/check-pr-title.yaml:39]: Oops, I should’ve thought to check for this in the Tamanu repo. I’d better update the Slab docs

**Review by jaskfla — APPROVED** [review]
Two comments, but both minor 🙏

File: README.md (hunk 1 above)
Comment (rohan-bes) [reply README.md:167]: Ah good point, will move it there

**Pull request merged**